File: android-native/app/src/main/java/com/example/kpkn/screens/home/HomeHeaderSection.kt
package com.example.kpkn.screens.home

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Notifications
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.kpkn.R
import com.example.kpkn.ui.components.icons.CaupolicanIcon

@Composable
fun HomeHeaderSection(
    modifier: Modifier = Modifier,
    greeting: String,
    userName: String,
    ringsViewMode: HomeViewModel.RingsViewMode,
    onThemeToggle: () -> Unit,
    onSettingsClick: () -> Unit,
    onRingsViewChange: (HomeViewModel.RingsViewMode) -> Unit,
) {
    Column(modifier.fillMaxWidth().padding(horizontal = 24.dp).padding(top = 16.dp, bottom = 20.dp)) {
        // Solo el saludo — "Tus RINGS" está en HomeRingsSection
        Text(
            "$greeting,\n$userName!",
            style = MaterialTheme.typography.displaySmall,
            fontWeight = FontWeight.Black,
            lineHeight = 40.sp,
            letterSpacing = (-1).sp,
        )
    }
}

// ─── Header Icons ──────────────────────────────────────────────────────────

@Composable
private fun ThreeRingsHeaderIcon(modifier: Modifier = Modifier) {
    Canvas(modifier.fillMaxSize()) {
        val size = this.size
        val r = size.minDimension / 6f
        val s = r * 1.6f
        val cy = size.height / 2f
        val cx = size.width / 2f

        val colors = listOf(Color(0xFFFF5252), Color(0xFF448AFF), Color(0xFFFFD740))
        val positions = listOf(
            Offset(cx - s, cy),
            Offset(cx, cy),
            Offset(cx + s, cy),
        )

        positions.forEachIndexed { i, center ->
            drawCircle(
                colors[i],
                r,
                center,
                style = Stroke(1.2f),
            )
        }
    }
}

@Composable
private fun SingleRingHeaderIcon(modifier: Modifier = Modifier) {
    Canvas(modifier.fillMaxSize()) {
        val size = this.size
        val r = size.minDimension / 2.5f
        val center = Offset(size.width / 2f, size.height / 2f)

        drawCircle(
            Color(0xFF448AFF),
            r,
            center,
            style = Stroke(1.2f),
        )
    }
}

File: ios-native/KPKNFit/KPKNFit/Presentation/Components/HomeHeaderSection.swift
import SwiftUI

/// Swift translation of HomeHeaderSection.kt
public struct HomeHeaderSection: View {
    let greeting: String
    let userName: String
    
    public init(greeting: String, userName: String) {
        self.greeting = greeting
        self.userName = userName
    }
    
    public var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            Text("\(greeting),\n\(userName)!")
                .font(.system(size: 36, weight: .black)) // displaySmall equivalent
                .lineSpacing(4)
                .tracking(-1)
                .foregroundColor(.white)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(.horizontal, 24)
        .padding(.top, 16)
        .padding(.bottom, 20)
    }
}

// ─── Header Icons ──────────────────────────────────────────────────────────

private struct ThreeRingsHeaderIcon: View {
    let colors = [
        Color(red: 1.0, green: 0.32, blue: 0.32), // 0xFFFF5252
        Color(red: 0.26, green: 0.54, blue: 1.0),  // 0xFF448AFF
        Color(red: 1.0, green: 0.84, blue: 0.25)   // 0xFFFFD740
    ]
    
    var body: some View {
        Canvas { context, size in
            let r = size.minDimension / 6.0
            let s = r * 1.6
            let cy = size.height / 2.0
            let cx = size.width / 2.0
            
            let centers = [
                CGPoint(x: cx - s, y: cy),
                CGPoint(x: cx, y: cy),
                CGPoint(x: cx + s, y: cy)
            ]
            
            for i in 0..<centers.count {
                let rect = CGRect(
                    x: centers[i].x - r,
                    y: centers[i].y - r,
                    width: r * 2.0,
                    height: r * 2.0
                )
                context.stroke(
                    Path(ellipseIn: rect),
                    with: .color(colors[i]),
                    lineWidth: 1.2
                )
            }
        }
    }
}

private struct SingleRingHeaderIcon: View {
    var body: some View {
        Canvas { context, size in
            let r = size.minDimension / 2.5
            let center = CGPoint(x: size.width / 2.0, y: size.height / 2.0)
            let rect = CGRect(
                x: center.x - r,
                y: center.y - r,
                width: r * 2.0,
                height: r * 2.0
            )
            context.stroke(
                Path(ellipseIn: rect),
                with: .color(Color(red: 0.26, green: 0.54, blue: 1.0)),
                lineWidth: 1.2
            )
        }
    }
}

#Preview {
    ZStack {
        Color.black.ignoresSafeArea()
        VStack {
            HomeHeaderSection(greeting: "Buenos días", userName: "Carlos")
            HStack(spacing: 20) {
                ThreeRingsHeaderIcon()
                    .frame(width: 48, height: 48)
                SingleRingHeaderIcon()
                    .frame(width: 48, height: 48)
            }
        }
    }
}
